feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -395,40 +395,35 @@ func TestComponentParent(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.desc, func(t *testing.T) {
-			t.Logf("Found card list for Component Type %v : %v", tc.componentType, cardList[tc.desc])
-			if len(cardList[tc.desc]) == 0 {
-				t.Fatalf("Get Card list for %q: got 0, want > 0", dut.Model())
+			t.Logf("Found Component list for Type %v : %v", tc.componentType, compList[tc.desc])
+			if len(compList[tc.desc]) == 0 {
+				t.Fatalf("Get Component list for %q: got 0, want > 0", dut.Model())
 			}
 			// Validate parent component
-			for _, card := range cardList[tc.desc] {
-				t.Logf("Validate card %s", card)
-				cardName := card
+			for _, comp := range compList[tc.desc] {
+				t.Logf("Validate component %s", comp)
+				compName := comp
+				foundChassis := false
 				for {
-					parent := gnmi.Lookup(t, dut, gnmi.OC().Component(cardName).Parent().State())
+					parent := gnmi.Lookup(t, dut, gnmi.OC().Component(compName).Parent().State())
 					val, present := parent.Val()
 					if !present {
-						t.Fatalf("Parent not present for %q: got %v, want true", card, parent.IsPresent())
+						t.Fatalf("Parent not present for %q: got %v, want true", comp, parent.IsPresent())
 					} else {
 						got := gnmi.Get(t, dut, gnmi.OC().Component(val).Type().State())
-						if tc.desc == "SwitchChip" || tc.desc == "FabricChip" {
-							if got == componentParent["FabricChip"] || got == componentParent["SwitchChip"] {
-								t.Logf("Got expected parent for card %s", card)
-								break
-							}
-						} else {
-							if got == tc.parent {
-								t.Logf("Got expected parent for card %s", card)
-								break
-							}
+						if got == chassisType {
+							foundChassis = true
+							t.Logf("Found chassis component in the hierarchy tree of component %s", comp)
+							break
 						}
-
 					}
-					parentName := gnmi.Get(t, dut, gnmi.OC().Component(val).Name().State())
-					cardName = parentName
+					// Not reached chassis yet; go one level up
+					compName = gnmi.Get(t, dut, gnmi.OC().Component(val).Name().State())
+				}
+				if !foundChassis {
+					t.Fatalf("Chassis component NOT found in the hierarchy tree of component %s", comp)
```

Comment:
Change to t.Errorf, so you get failures for multiple components in a single test run